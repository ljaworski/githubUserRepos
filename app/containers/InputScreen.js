import React, { useState } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { View } from "react-native";
import { Button, Input, HeaderTitle } from "../components";
import { fetchRepos } from "../actions/reposActions";
import { InputScreenStyle, DarkModeStyle } from "../style";

export function InputScreen({ navigation, fetchGithubRepos, darkMode }) {
  const [username, setUsername] = useState("");
  const getRepos = () => {
    if (username.length) {
      setUsername("");
      fetchGithubRepos(username);
      navigation.navigate("List");
    }
  };
  return (
    <View
      style={[
        InputScreenStyle.container,
        darkMode ? DarkModeStyle.container : null
      ]}
    >
      <View style={InputScreenStyle.input}>
        <Input
          onChange={setUsername}
          value={username}
          placeholder="Github username"
          style={darkMode ? DarkModeStyle.input : null}
        />
      </View>
      <Button
        label="Find most successful repository"
        onPress={getRepos}
        disabled={false}
        style={InputScreenStyle.button}
      />
    </View>
  );
}

InputScreen.navigationOptions = {
  headerTitle: <HeaderTitle title="Github Input" />
};

InputScreen.propTypes = {
  darkMode: PropTypes.bool,
  fetchGithubRepos: PropTypes.func
};

const mapStateToProps = ({ appSettings: { darkMode } }) => ({
  darkMode
});

const mapDispatchToProps = dispatch => ({
  fetchGithubRepos: username => dispatch(fetchRepos(username))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(InputScreen);
