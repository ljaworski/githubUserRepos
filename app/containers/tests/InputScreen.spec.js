import React from "react";
import renderer from "react-test-renderer";
import { InputScreen } from "../InputScreen";

const props = {
  navigation: {
    navigate: () => {}
  },
  fetchGithubRepos: () => {},
  darkMode: false
};

test("InputScreen renders correctly", () => {
  const tree = renderer.create(<InputScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
