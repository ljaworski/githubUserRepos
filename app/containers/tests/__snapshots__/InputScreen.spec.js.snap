// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputScreen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      null,
    ]
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 1,
          "borderRadius": undefined,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 2,
          "opacity": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="Github username"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "paddingLeft": 8,
              "paddingRight": 5,
              "top": 1.5,
            },
            null,
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Find most successful repository
    </Text>
  </View>
</View>
`;
